### PR TITLE
JS SDK MAPI VNext verification

### PR DESCRIPTION
**Pull request by Simply007**
Testing the new vNext version of  MAPI SDK - 1.0.0-next.0.

### CODE REVIEW

**Comment by Simply007** [comment]
I don!T know how to solve the issue:


```plain
Could not find a declaration file for module '@kentico/kontent-management'. 'c:/projects/kontent-migrations-boilerplate/node_modules/@kentico/kontent-management/dist/_bundles/kontent-management.umd.min.js' implicitly has an 'any' type.
 Try `npm i --save-dev @types/kentico__kontent-management` if it exists or add a new declaration (.d.ts) file containing `declare module '@kentico/kontent-management';`
```

Command `npm i --save-dev @types/kentico__kontent-management` fails, because there is no such package -  so the MAPI JS SDK package itself should probably include the declaration file

**Comment by Enngage** [comment]
@Simply007  thank you :) I've fixed typings & modules in the `1.0.0-next.3`. Should be working now ;)

**Comment by Simply007** [comment]
@Enngage  - I don't know why but when I extract builder for Migration `01` and `03` the builder types seems to be incompatible with `ContentTypeElementsBuilder`.

https://github.com/Kentico/kontent-migrations-boilerplate/pull/3/checks?check_run_id=2795036971#step:6:8

File: Migrations/02_sample_init_createBlogItem.ts [hunk 1]
```diff
@@ -22,29 +22,29 @@ const migration: MigrationModule = {
             .upsertLanguageVariant()
             .byItemId(itemResponse.data.id)
             .byLanguageCodename('default')
-            .withElements([
-                {
+            .withData((builder) => [
+                builder.textElement({
```
Comment (Simply007) [inline Migrations/02_sample_init_createBlogItem.ts:26]: @Enngage - is there any preference for building elements - with/without a builder?

**Comment by Simply007** [comment]
> @Enngage - I don't know why but when I extract builder for Migration `01` and `03` the builder types seems to be incompatible with `ContentTypeElementsBuilder`.
> 
> https://github.com/Kentico/kontent-migrations-boilerplate/pull/3/checks?check_run_id=2795036971#step:6:8
> 
> The error states also:
> 
> ```
> The types returned by 'contentGroup(...)' are incompatible between these types.
>         Property 'type' is missing in type 
> 'import("c:/projects/kontent-migrations-boilerplate/node_modules/@kentico/kontent-cli/node_modules/@kentico/kontent-management/_commonjs/models/content-types/content-type-elements.builder").ContentTypeElements.IElementInContentType'
> but required in type
> 'import("c:/projects/kontent-migrations-boilerplate/node_modules/@kentico/kontent-management/dist/cjs/models/content-types/content-type-elements.builder").ContentTypeElements.IElementInContentType'
> ```
> 
> I my opinion, the Kontent-management SDK might be the peer dependency of Kontent CLI.

Let's make MAPI JS SDK a peer dependency and release a beta and test it out.

**Comment by Simply007** [comment]
@Enngage - I have used Kontent CLI with Mapi SDK as Peer dependency (and using `next` version here).

There are still some CI errors - the most interesting is this one: https://github.com/Kentico/kontent-migrations-boilerplate/pull/3/checks?check_run_id=2876540519#step:6:13

`Error: node_modules/@kentico/kontent-cli/lib/types/index.d.ts(4,20): error TS2314: Generic type 'IManagementClient<TCancelToken>' requires 1 type argument(s).`

We have used `ManagementClient` as a Module parameter`- do you thing it might cause this issue?

**Comment by Enngage** [comment]
@Enngage you should use `ManagementClient` class instead of the interface across the project :) The error you mentioned should be fixed here -> https://github.com/Kentico/kontent-cli/blob/feature/21/src/types/index.ts#L5

**Comment by Simply007** [comment]
> @Enngage you should use `ManagementClient` class instead of the interface across the project :) The error you mentioned should be fixed here -> https://github.com/Kentico/kontent-cli/blob/feature/21/src/types/index.ts#L5

Fixed

---

I will go through the sample migration and test some other scenarios.

**Comment by Simply007** [comment]
Scenarios work well.

Wait for https://github.com/Kentico/kontent-cli/pull/24 before the merge.

**Review by winklertomas — APPROVED** [review]

**Pull request merged**